nautobot_golden_config/utilities/config_to_push.py

```diff
@@ -0,0 +1,162 @@
+"""Functions related to prepare configuration to push."""
+from functools import partial
+from typing import Optional
+import jinja2
+from jinja2 import exceptions as jinja_errors
+
+from django.http import HttpRequest
+from django.core.exceptions import ObjectDoesNotExist
+
+from nautobot.dcim.models import Device
+from nautobot.users.models import User
+from nautobot.extras.models.secrets import SecretsGroup
+from nautobot.extras.choices import SecretsGroupAccessTypeChoices
+
+from netutils.utils import jinja2_convenience_function
+
+from nautobot_golden_config import models
+from nautobot_golden_config.utilities.constant import PLUGIN_CFG, ENABLE_PUSH
+from nautobot_golden_config.utilities.graphql import graph_ql_query
+from nautobot_golden_config.exceptions import RenderConfigToPushError
+from nautobot_golden_config.utilities.helper import get_device_to_settings_map
+
+
+def get_secret_by_secret_group_slug(
+    user: User,
+    secrets_group_slug: str,
+    secret_type: str,
+    secret_access_type: Optional[str] = SecretsGroupAccessTypeChoices.TYPE_GENERIC,
```

Comment:
seems odd to do typing on just random pieces of code.

Reply:
I tried to do in all this PR, may I have left something out?